test/test_tr_spectra.py

```diff
@@ -0,0 +1,134 @@
+import os
+import sys
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy as sp
+import typing
+
+from icecream import ic
+
+sys.path.append('.')
+sys.path.append('./scripter/')
+sys.path.append('./commander/')
+
+if os.environ.get('CORELOOP_DIR') is not None:
+    sys.path.append(os.environ.get('CORELOOP_DIR'))
+
+from test_base import Test
+from lusee_script import Scripter
+from collections import defaultdict
+
+from pycoreloop import appId
+
+import uncrater as uc
+
+
+class Test_TRSpectra(Test):
+    name = "time_resolved"
+    version = 0.1
+    description = """ Check that time resolved spectra are received."""
+    instructions = """ Do not need to connect anything."""
+    default_options = {
+        "time": 10,
+        "tr_start": 0,
+        "tr_stop": 8,
+        "tr_avg_shift": 1
+    }  ## dictinary of options for the test
+    options_help = {
+        "time": "Total time to run the test.",
+        "tr_start": "Start of time-resolved window.",
+        "tr_stop": "End of time-resolved window.",
+        "tr_avg_shift": "Average over every 2^tr_avg_shift values.",
+    }  ## dictionary of help for the options
+
+    def generate_script(self):
+        """ Generates a script for the test """
+        if self.time < 20:
+            print("Time raised to 20 seconds.")
+            self.time = 20
+
+        scripter = Scripter()
+        scripter.start()
+        # set Navg_1 and Navg_2 to 4 (2^2 -- we send shifts)
+        # to produces lots of spectra
+        scripter.set_Navg(3, 3)
+        scripter.set_tr_start_lsb(self.tr_start)
+        scripter.set_tr_stop_lsb(self.tr_stop)
+        scripter.set_tr_avg_shift(self.tr_avg_shift)
+        scripter.wait(self.time)
+        scripter.stop()
+        # scripter.time_to_die()
+        return scripter
+
+    def analyze(self, C, uart, commander, figures_dir):
+        """ Analyzes the results of the test.
+            Returns true if test has passed.
+        """
+        self.results = {}
+        passed = True
+
+        self.results['packets_received'] = len(C.cont)
+
+        C.cut_to_hello()
+
+        if len(C) > 0 and type(C.cont[0]) == uc.Packet_Hello:
```

Comment:
Please use `self.get_versions` defined in base class that has a copy of this code since it gets used everywhere.

Reply:
Done.